src/data/trace.ts

```diff
@@ -133,12 +133,16 @@ export const loadTraceContexts = (
   hass: HomeAssistant,
   domain?: string,
   item_id?: string
-): Promise<TraceContexts> =>
-  hass.callWS({
-    type: "trace/contexts",
-    domain,
-    item_id,
-  });
+): Promise<TraceContexts> | undefined => {
+  if (hass.user?.is_admin) {
+    return hass.callWS({
+      type: "trace/contexts",
+      domain,
+      item_id,
+    });
+  }
+  return undefined;
```

Comment:
I think we should not call this function if not an admin instead, we can't return undefined, at least not with changing more. `ha-logbook` can't handle `undefined` for `traceContexts`.

Reply:
I might be missing your point, but moving the is-admin check out of `loadTraceContexts()` will not really change something. In either way we end up with no trace contexts. All usages of `loadTraceContexts()` eventually result in a `ha-logbook` being rendered so we always have to provide something for the `traceContexts`. In `ha-logbook` currently `traceContexts` is defaulted to `{}` so we could return that instead from `loadTraceContexts()`.

Reply:
Thought about it some more and I think I now see what you meant. Commit coming in a minute.